compiler/rustc_borrowck/src/lib.rs

```diff
@@ -300,18 +300,14 @@ fn do_mir_borrowck<'tcx>(
             move_errors: Vec::new(),
             diags,
         };
-        MoveVisitor { ctxt: &mut promoted_mbcx }.visit_body(promoted_body);
+        promoted_mbcx.visit_body(promoted_body);
         promoted_mbcx.report_move_errors();
         diags = promoted_mbcx.diags;
 
-        struct MoveVisitor<'a, 'b, 'mir, 'infcx, 'tcx> {
-            ctxt: &'a mut MirBorrowckCtxt<'b, 'mir, 'infcx, 'tcx>,
-        }
-
-        impl<'tcx> Visitor<'tcx> for MoveVisitor<'_, '_, '_, '_, 'tcx> {
+        impl<'tcx> Visitor<'tcx> for MirBorrowckCtxt<'_, '_, 'tcx> {
```

Comment:
This impl is now accessible from anywhere, and meaningless or wrong for everything but this one usage site. Please keep the previous wrapper struct

Reply:
Ok, I removed this commit.